src/components/ec/cuda/ec_cuda_executor_interruptible.c

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020-2022.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "ec_cuda_executor.h"
+#include "components/mc/ucc_mc.h"
+
+ucc_status_t ucc_cuda_executor_interruptible_get_stream(cudaStream_t *stream)
+{
+    static int last_used = 0;
+    cudaError_t cuda_st;
+    int i, j;
+
+    ucc_spin_lock(&ucc_ec_cuda.init_spinlock);
+    if (ucc_unlikely(ucc_ec_cuda.exec_streams[0] == NULL)) {
+        for(i = 0; i < EC_CUDA_CONFIG->exec_num_streams; i++) {
+            cuda_st = cudaStreamCreateWithFlags(&ucc_ec_cuda.exec_streams[i],
```

Comment:
will it be optimal to create steam on demand at line:35 on first use?
I think cuda stream is a costly object. if I remember it consumes ~100 MB GPU memory.  can we check this again?

Reply:
I don't think it makes sense, anyway after 16 ops all streams will be initialized because we use them one by one. Similar initialization scheme is used in cuda_ipc in UCX https://github.com/openucx/ucx/blob/b054f883c9f8912444f34a48873429c9041c4f03/src/uct/cuda/cuda_ipc/cuda_ipc_iface.c#L451